lib/ansible/modules/cloud/vmware/vmware_drs_group.py

```diff
@@ -0,0 +1,600 @@
+    hosts:
+      - DC0_C0_H0
+      - DC0_C0_H1
+      - DC0_C0_H2
+
+- name: "Delete DRS Host group"
+  delegate_to: localhost
+  state: absent
+  vmware_drs_group:
+    hostname: "{{ vcenter_hostname }}"
+    password: "{{ vcenter_password }}"
+    username: "{{ vcenter_username }}"
+    cluster_name: DC0_C0
+    datacenter_name: DC0
+    group_name: TEST_HOST_01
+
+'''
+
+RETURN = r'''
+drs_group_facts:
+    description: Metadata about DRS group created
+    returned: always
+    type: dict
+    sample:
+        "drs_group_facts": {
+        "changed": true,
+        "failed": false,
+        "msg": "Created host group TEST_HOST_01 successfully",
+        "result": {
+            "DC0_C0": [
+                {
+                    "group_name": "TEST_HOST_01",
+                    "hosts": [
+                        "DC0_C0_H0",
+                        "DC0_C0_H1",
+                        "DC0_C0_H2"
+                    ],
+                    "type": "host"
+                }
+            ]
+        }
+    }
+'''
+
+try:
+    from pyVmomi import vim
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import (PyVmomi, vmware_argument_spec,
+                                         wait_for_task, find_cluster_by_name,
+                                         find_vm_by_id, find_datacenter_by_name, find_vm_by_name)
+
+
+class VmwareDrsGroupManager(PyVmomi):
+    """
+    Class to manage DRS groups
+    """
+
+    def __init__(self, module, cluster_name, group_name, state,
+                 datacenter_name=None, vm_list=None, host_list=None):
+        """
+        Init
+        """
+
+        super(VmwareDrsGroupManager, self).__init__(module)
+
+        self.__datacenter_name = datacenter_name
+        self.__datacenter_obj = None
+        self.__cluster_name = cluster_name
+        self.__cluster_obj = None
+        self.__group_name = group_name
+        self.__vm_list = vm_list
+        self.__vm_obj_list = []
+        self.__host_list = host_list
+        self.__host_obj_list = []
+        self.__msg = 'Nothing to see here...'
+        self.__result = dict()
+        self.__changed = False
+        self.__state = state
+
+        if datacenter_name is not None:
+
+            self.__datacenter_obj = find_datacenter_by_name(self.content, self.__datacenter_name)
+
+            if self.__datacenter_obj is None and module.check_mode is False:
+                raise Exception("Datacenter '%s' not found" % self.__datacenter_name)
+
+        self.__cluster_obj = find_cluster_by_name(content=self.content,
+                                                  cluster_name=self.__cluster_name,
+                                                  datacenter=self.__datacenter_obj)
+
+        # Throw error if cluster does not exist
+        if self.__cluster_obj is None and module.check_mode is False:
+            raise Exception("Cluster '%s' not found" % self.__cluster_name)
```

Comment:
Same as above, please fail the module, instead of raising an exception
```suggestion
            self.module.fail_json(msg="Cluster '%s' not found" % self.__cluster_name)
```